packages/monaco/src/EditorHooks.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+export function useResetEditor(): { add: (cb: () => void) => void; dispatch: () => void } {
+  const ev = new CustomEvent('resetCode');
+
+  function add(cb: () => void) {
```

Comment:
i think a more appropriate name is subscribe.

out of scope but could probably make this more generic by being able to pass event name

Reply:
I agree with naming it subscribe but I question if we even should follow this or use zustand and create a store as we already have it as a dep. 

Depending on the other editor functionalities. But as these are just signals _I think_ a simple solution here is sufficient.

Reply:
I changed it to be more generic. I don't know if this is common to setup like this or if hooks being factories is more common.